diff --git a/dist/index.js b/dist/index.js
index 54e1bd29d062ad8cab1568bf43905ee3371faa87..515bc106d51aeace1b4301d7306e51baf3b3aea3 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -748,14 +748,16 @@ var addUnlinkedScrollListener = /* @__PURE__ */ __name((node, handler = () => {
 }) => {
   let prevPosition = { left: node.scrollLeft, top: node.scrollTop };
   let rAF = 0;
-  (/* @__PURE__ */ __name((function loop() {
+  function loop() {
     const position = { left: node.scrollLeft, top: node.scrollTop };
     const isHorizontalScroll = prevPosition.left !== position.left;
     const isVerticalScroll = prevPosition.top !== position.top;
     if (isHorizontalScroll || isVerticalScroll) handler();
     prevPosition = position;
     rAF = window.requestAnimationFrame(loop);
-  }), "loop"))();
+  }
+  __name(loop, "loop");
+  loop();
   return () => window.cancelAnimationFrame(rAF);
 }, "addUnlinkedScrollListener");
 function useDebounceCallback(callback, delay) {
diff --git a/dist/index.mjs b/dist/index.mjs
index b98d4bc14ad4184b2f46b28679fb5f8d55767782..4662ba3050eccd6711a508769fc6f42f6e33706e 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -704,14 +704,16 @@ var addUnlinkedScrollListener = /* @__PURE__ */ __name((node, handler = () => {
 }) => {
   let prevPosition = { left: node.scrollLeft, top: node.scrollTop };
   let rAF = 0;
-  (/* @__PURE__ */ __name((function loop() {
+  function loop() {
     const position = { left: node.scrollLeft, top: node.scrollTop };
     const isHorizontalScroll = prevPosition.left !== position.left;
     const isVerticalScroll = prevPosition.top !== position.top;
     if (isHorizontalScroll || isVerticalScroll) handler();
     prevPosition = position;
     rAF = window.requestAnimationFrame(loop);
-  }), "loop"))();
+  }
+  __name(loop, "loop");
+  loop();
   return () => window.cancelAnimationFrame(rAF);
 }, "addUnlinkedScrollListener");
 function useDebounceCallback(callback, delay) {
